faucet/frontend/Dockerfile

```diff
@@ -1,7 +1,10 @@
 FROM node:lts-alpine AS builder
 WORKDIR /app
 COPY . .
-RUN npm install && npm run build
-RUN npm install -g serve
+RUN npm install && npm install -g serve
+
+# Build the app to cache the dependencies
+RUN npm run build
```

Comment:
The npm run build can be on the same RUN line as the npm install since the build should not change unless deps change.